linalg/operator.hpp

```diff
@@ -309,30 +309,37 @@ class Operator
 
 
 /// Base abstract class for first order time dependent operators.
-/** Operator of the form: (x,t) -> f(x,t), where k = f(x,t) generally solves the
-    algebraic equation F(x,k,t) = G(x,t). The functions F and G represent the
-    _implicit_ and _explicit_ parts of the operator, respectively. For explicit
-    operators, F(x,k,t) = k, so f(x,t) = G(x,t). */
+/** Operator of the form: (u,t) -> k(u,t), where k generally solves the
+    algebraic equation F(u,k,t) = G(u,t). The functions F and G represent the
+    _implicit_ and _explicit_ parts of the operator, respectively.
+
+    A common use for this class is representing a differential algebraic
+    equation of the form \f$ F(y,\frac{dy}{dt},t) = G(y,t) \f$. Consider an
+    ordinary differential equation of the form \f$ M \frac{dy}{dt} = g(y,t) \f$,
+    where F and G are defined as one of the following:
+    1. F(u,k,t) = k and G(u,t) = inv(M) g(u,t)
+    2. F(u,k,t) = M k and G(u,t) = g(u,t)
+    */
 class TimeDependentOperator : public Operator
 {
 public:
    enum Type
    {
-      EXPLICIT,   ///< This type assumes F(x,k,t) = k, i.e. k = f(x,t) = G(x,t).
+      EXPLICIT,   ///< This type assumes F(u,k,t) is linear in k.
       IMPLICIT,   ///< This is the most general type, no assumptions on F and G.
-      HOMOGENEOUS ///< This type assumes that G(x,t) = 0.
+      HOMOGENEOUS ///< This type assumes that G(u,t) = 0.
    };
 
    /// Evaluation mode. See SetEvalMode() for details.
    enum EvalMode
    {
       /** Normal evaluation. */
       NORMAL,
-      /** Assuming additive split, f(x,t) = f1(x,t) + f2(x,t), evaluate the
-          first term, f1. */
+      /** Assuming additive split, G(u,t) = G1(u,t) + G2(u,t), evaluate with
+          F(u,k,t) = G1(u,t). */
       ADDITIVE_TERM_1,
-      /** Assuming additive split, f(x,t) = f1(x,t) + f2(x,t), evaluate the
-          second term, f2. */
+      /** Assuming additive split, G(u,t) = G1(u,t) + G2(u,t), evaluate with
+          F(u,k,t) = G2(u,t). */
```

Comment:
This is also changing meaning in a way that is not backward compatible.

Reply:
Can you elaborate?  When solving $M \frac{dy}{dt} = g_1(y,t) + g_2(y,t)$, where $F(u,k,t) = M k$ and $G(u,t) = g_1(u,t) + g_2(u,t)$, wouldn't an additive split method solve either $M \frac{dy}{t} = g_1(y,t)$ or $M \frac{dy}{t} = g_2(y,t)$?

Reply:
What I'm thinking about is the general case where $F(u,k,t)$ is NOT linear in $k$. In that case assuming additive splitting for $f(u,t)$ is not the same as assuming additive splitting for $G(u,t)$. Basically, assuming the relations: $$F(u,f(u,t),t) = G(u,t), \quad F(u,f_1(u,t),t) = g_1(u,t), \quad F(u,f_2(u,t),t) = g_2(u,t)$$ then:

1. if $G(u,t) = g_1(u,t) + g_2(u,t)$ then, in general, $f(u,t) \ne f_1(u,t) + f_2(u,t)$;
2. and vice-versa: if $f(u,t) = f_1(u,t) + f_2(u,t)$, then, in general $G(u,t) \ne g_1(u,t) + g_2(u,t)$.

If this type of additive split is never used when $F(u,k,t)$ is NOT linear in $k$, then we can state that assumption (F is linear in k) for all methods and type-constants related to the additive split -- and then an additive split for $f$ is equivalent to an additive split of `G`, so the documentation changes will be backward-compatible (with the previously-not-mentioned assumption that F is linear in k 😁).

Reply:
I'm not clear on what the $f(u,t)$ is.  In the following DAE: $y \frac{dy}{dt} = g_1(y,t) + g_2(y,t)$, what is the $f(u,t)$?

Reply:
Following the definitions in the class doxygen $f(u,t)$ is the solution to $F(u,f(u,t),t) = G(u,t)$, so in the equation you asked about (I'm not sure why you call it DAE, to me this is just an ODE), $f(y,t)$ (you ask about $f(u,t)$ but I'm not sure if there is a difference) is simply $f(y,t) = (1/y)(g_1(y,t)+g_2(y,t)$, so the ODE is simply $\frac{dy}{dt} = f(y,t)$.

Reply:
Note that the documentation was written exclusively with ODEs in mind, so it was assumed that $F(u,f(u,t),t)=G(u,t)$ can be inverted and solved for $f(u,t)$. In the case when that is not true and some equations in $F(u,k,t)=G(u,t)$ are independent of $k$, then proper generalizations must be considered and the documentation updated carefully.

Reply:
>I'm not sure why you call it DAE, to me this is just an ODE

Yeah good point.  I think I have what I need to take another stab at this.  Stay tuned.

Reply:
I just pushed a commit where I opted to just stick with the original definition with the new notation.

Reply:
I believe this is resolved, please feel free to re-open this thread if it is not.